lib/ocpp/common/database/database_connection.cpp

```diff
@@ -28,14 +33,22 @@ bool DatabaseConnection::open_connection() {
         EVLOG_error << "Error opening database at " << this->database_file_path << ": " << sqlite3_errmsg(db);
         return false;
     }
-    EVLOG_info << "Established connection to Database: " << this->database_file_path;
+    EVLOG_info << "Established connection to database: " << this->database_file_path;
     return true;
 }
 
 bool DatabaseConnection::close_connection() {
-    EVLOG_info << "Closing database connection...";
+    return this->close_connection_internal(false);
+}
+
+bool DatabaseConnection::close_connection_internal(bool force_close) {
+    if (!force_close && this->open_count.fetch_sub(1) != 1) {
+        EVLOG_debug << "Connection should remain open for other users";
+        return true;
```

Comment:
Shouldn't this return false since the closing did not happen?

Reply:
My idea behind true was that the operation did not fail per se. If I would return false then the user of the function might think it failed.